Fix/ranked search select

duk3luk3 opened the pull request:
Potential fix for #548.

The race selection was buggy and did not register the first click
properly.
The stringly typing was removed and that fixed the issue, somehow.

- [ ] PR branch should be named `issuenum`-`fix`/`feature`/`cleanup`-`description`
- [ ] Code is split into logical commits
- [ ] Code has tests
- [ ] Final commit includes "Fixes #issue" in commit message

When all builds pass and a maintainer is happy with your PR, the "ready" label will be applied. Please complete these tasks then:

- [ ] Rebase onto develop
- [ ] Add changelog entry
- [ ] Remove this entire template section

Wesmania committed:
Add resources directory logic

Add a function in a separate file (with no dependencies) to use
to locate the 'res' dir, depending whether we're frozen on Windows,
installed on unix or run from source.

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Replace/remove directory setting and finding logic from main and util

Now that we don't need a specific working directory and the resources
directory finding is extracted into a module, we can remove and replace
the logic in util and main.

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

duk3luk3 committed:
Make stderr log level adjustable

GrotheFAF commented:
look complicated, I just added an init line in _gamewidget.py L55:
 Settings.set("play/subFactions", self.sub_factions)  # needs one write before self.sub_factions... work

Wesmania committed:
Add res cleanup to changelog

This fixes #552.

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

duk3luk3 committed:
Global ssl cert verification disable

Turns off ssl certificate verification globally when in development environment
to circumvent broken test server config

Wesmania committed:
Rework version.py to be directory-agnostic

version.py is used both by install scripts to find and write the version
file and by FAF to read it. Have version finding functions explicitly
accept directories where they might find the version file and modify
install scripts and FAF to provide them.

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

Wesmania committed:
Move RELEASE-VERSION to res directory

Signed-off-by: Igor Kotrasinski <ikotrasinsk@gmail.com>

duk3luk3 committed:
Add logging to updater download

duk3luk3 committed:
Add changelog for #558

duk3luk3 committed:
Fix ladder game search race selection

The race selection was buggy and did not register the first click
properly.
The stringly typing was removed and that fixed the issue.
